### PR TITLE
test: add unit tests to boost coverage to 95%

### PR DESCRIPTION
**Pull request by leseb**
## Summary

- Add unit tests across 12 files to increase line coverage from ~94.5% to 95.35%
- Cover previously untested code paths: error handling, edge cases, Display impls, pure helper functions
- Fix missing assert messages in guardrails tests (from #323 review feedback)

### Files changed
- `apis/src/anthropic/to_openai/{request,response,mod}.rs` — conversion error paths, tool_choice branches, metadata extraction
- `apis/src/openai/sse/{frame,responses/event}.rs` — SseParseError Display, event_type round-trip
- `apis/src/openai/conversations/{handlers,filter}.rs` — store_error_response, parse_item_list_params, reject_store_unavailable
- `apis/src/openai/responses/proxy/tests.rs` — strip_conversation_field edge cases
- `apis/src/store/tests.rs` — StoreError Display + Error trait
- `filters/src/inference/model_to_header.rs` — response-side delegate methods
- `filters/src/guardrails/tests.rs` — add missing assert messages
- `server/src/commands.rs` — default_config_source

## Test plan

- [x] `cargo test --workspace --lib` — 2035 tests pass
- [x] `make lint` — clippy + nightly rustfmt clean
- [x] `cargo llvm-cov` — coverage at 95.35%

### CODE REVIEW

File: server/src/commands.rs [hunk 1]
```diff
@@ -161,4 +161,19 @@ filter_chains:
             "error should mention the missing chain name: {err}"
         );
     }
+
+    #[test]
+    fn default_config_source_returns_builtin_when_no_yaml() {
+        let dir = tempfile::tempdir().unwrap();
+        let original = std::env::current_dir().unwrap();
+        std::env::set_current_dir(dir.path()).unwrap();
```
Comment (aslakknutsen) [inline server/src/commands.rs]: calls `std::env::set_current_dir` manually without the `CWD_MUTEX` + `CwdGuard` RAII pattern established in server/src/watcher.rs:475-493 for the same concern. A panic before restore could leave cwd wrong; parallel tests touching relative `praxis.yaml` paths (server.rs:484, watcher.rs:422+) have a narrow race window. Recommend reusing watcher test utilities or a shared cwd guard.

Comment (leseb) [reply server/src/commands.rs]: 236968ea9aec73fa067e494bbc016ddc31662030

**Review by aslakknutsen — APPROVED** [review]
